diff -u -r --new-file observium-trunk/html/api/v0/includes/oxidized.inc.php root/html/api/v0/includes/oxidized.inc.php
--- observium-trunk/html/api/v0/includes/oxidized.inc.php	1970-01-01 10:00:00.000000000 +1000
+++ root/html/api/v0/includes/oxidized.inc.php	2019-03-08 09:55:41.396331094 +1000
@@ -0,0 +1,28 @@
+<?php
+
+/**
+ * Observium
+ *
+ *   This file is part of Observium.
+ *
+ * @package    observium
+ * @subpackage api
+ * @author     Adam Armstrong <adama@observium.org>
+ * @copyright  (C) 2006-2013 Adam Armstrong, (C) 2013-2019 Observium Limited
+ *
+ */
+
+
+Flight::route('GET /oxidized',               'api_get_oxidized');
+
+function api_get_oxidized()
+{
+  $query = "SELECT hostname, os, type FROM `devices`; ";
+
+  $devices_db = dbFetchRows($query);
+
+  Flight::json($devices_db);
+
+}
+
+// EOF
diff -u -r --new-file observium-trunk/html/api/v0/index.php root/html/api/v0/index.php
--- observium-trunk/html/api/v0/index.php	2019-02-15 14:43:02.877666208 +1000
+++ root/html/api/v0/index.php	2019-03-08 09:42:58.539953730 +1000
@@ -52,6 +52,7 @@
 include("includes/alerts.inc.php");
 include("includes/bills.inc.php");
 include("includes/devices.inc.php");
+include("includes/oxidized.inc.php");
 include("includes/ports.inc.php");
 include("includes/sensors.inc.php");
 include("includes/status.inc.php");
--- observium-trunk/includes/definitions/transports.inc.php	2019-03-08 10:02:26.182308864 +1000
+++ root/includes/definitions/transports.inc.php	2019-03-12 10:19:38.778836530 +1000
@@ -130,6 +130,20 @@
   )
 );

+$config['transports']['oxidized'] = array(
+  'name' => 'Oxidized',
+  'identifiers' => array('url'),
+  'parameters' => array(
+    'required' => array(
+      'url' => array('description' => 'URL', 'tooltip' => "URL to Oxidized server, e.g. https://oxidized.your.tld"),
+    ),
+    'global' => array(
+      'header_name' => array('description' => 'Auth header name', 'tooltip' => 'Header name, e.g. Authorization'),
+      'header_value' => array('description' => 'Auth header value', 'tooltip' => 'Header value, e.g. Basic YWRtaW46RklYTUU='),
+    ),
+  )
+);
+
 $config['transports']['redoxygen'] = array(
   'name' => 'Red Oxygen',
   'identifiers' => array('acctid', 'email'),
diff -u -r --new-file observium-trunk/includes/alerting/oxidized.inc.php root/includes/alerting/oxidized.inc.php
--- observium-trunk/includes/alerting/oxidized.inc.php	1970-01-01 10:00:00.000000000 +1000
+++ root/includes/alerting/oxidized.inc.php	2019-03-12 10:00:20.040423045 +1000
@@ -0,0 +1,30 @@
+<?php
+
+/**
+ * Observium
+ *
+ *   This file is part of Observium.
+ *
+ * @package    observium
+ * @subpackage alerting
+ * @copyright  (C) 2006-2013 Adam Armstrong, (C) 2013-2015 Observium Limited
+ *
+ */
+
+
+if (isset($endpoint['url']) && $endpoint['url'] != '') {
+  $url = $endpoint['url'] . '/node/next/' . $message_tags['DEVICE_HOSTNAME'];
+  $context_data = array(
+    'method' => 'GET',
+    'header' =>
+      "Connection: close\r\n"
+  );
+
+  if (isset($endpoint['header_name']) && $endpoint['header_name'] != '' && isset($endpoint['header_value'])) {
+    $context_data['header'][] = $endpoint['header_name'] . ": " . $endpoint['header_value'] . "\r\n";
+  }
+
+  get_http_request($url, $context_data);
+}
+
+// EOF
--- observium-trunk/includes/defaults.inc.php	2019-03-12 10:16:27.934447072 +1000
+++ root/includes/defaults.inc.php	2019-03-12 18:51:20.754345397 +1000
@@ -521,6 +522,12 @@
 $config['rancid_revisions']             = 10; // Show such last count revisions in device page
 #$config['collectd_dir']                 = '/var/lib/collectd/rrd';

+// Oxidized Web Integration
+$config['oxidized']['web']['enable']         = FALSE; // Disable by default, user needs to define
+$config['oxidized']['web']['url']            = 'http://127.0.0.1:8888'; // Default is on same box
+$config['oxidized']['web']['auth']['header'] = ''; // e.g. Authorization
+$config['oxidized']['web']['auth']['value']  = ''; // e.g. Basic BASE64STRING
+
 // Smokeping
 #$config['smokeping']['dir']             = "/var/lib/smokeping/";
 $config['smokeping']['split_char']      = "_";
--- observium-trunk/html/pages/device/showtech.inc.php	2019-02-15 14:43:01.948652515 +1000
+++ root/html/pages/device/showtech.inc.php	2019-03-13 22:43:38.915463867 +1000
@@ -79,6 +79,54 @@

   echo generate_box_close();

+  $box_args = array('title' => 'Oxidized',
+                    'header-border' => TRUE,
+                    'padding' => TRUE,
+                   );
+
+  $box_args['header-controls'] = array('controls' => array('perf' => array('text' => 'Show Config',
+                                                                           'anchor' => TRUE,
+                                                                           'url'  => generate_url(array('page' => 'device', 'device' => $device['device_id'], 'tab' => 'showconfig')),
+                                                                          )));
+
+  // Oxidized Integration
+  if (isset($config['oxidized']['web']['enable'])
+    && $config['oxidized']['web']['enable']
+    && isset($config['oxidized']['web']['url'])
+    && $config['oxidized']['web']['url'] != '')
+  {
+    // do requests to Oxidized API
+    $context_data = array(
+      'method' => 'GET',
+      'header' =>
+        "Accept: application/json\r\n"
+    );
+
+    if (isset($config['oxidized']['web']['auth']['header'])
+        && isset($config['oxidized']['web']['auth']['value'])
+        && $config['oxidized']['web']['auth']['header'] != ''
+        && $config['oxidized']['web']['auth']['value'] != '')
+    {
+      $context_data['header'][] = $config['oxidized']['web']['auth']['header'] . ": " . $config['oxidized']['web']['auth']['value'] . "\r\n";
+    }
+
+    echo generate_box_open($box_args);
+
+    $url = $config['oxidized']['web']['url'] . '/node/show/' . $device['hostname'] . '?format=json';
+    $result = json_decode(get_http_request($url, $context_data), TRUE);
+
+    if ($result)
+    {
+      echo('<p />');
+
+      print_success("Configuration for device was found; will be displayed to users with level 7 or higher.");
+    } else {
+      print_warning("Oxidized does not appear to monitor this device.");
+    }
+
+    echo generate_box_close();
+  }
+
   $box_args = array('title' => 'Smokeping',
                     'header-border' => TRUE,
                     'padding' => TRUE,
--- observium-trunk/html/pages/device.inc.php	2019-02-15 14:43:01.961652707 +1000
+++ root/html/pages/device.inc.php	2019-03-13 22:44:17.690954801 +1000
@@ -476,10 +476,14 @@
     // If the user has secure global read privileges, check for a device config.
     if ($_SESSION['userlevel'] >= 7)
     {
+      if ($config['oxidized']['web']['enable']) {
+        $oxidized_hostpath = $device['type'] . '/' . $device['hostname'];
+      }
+
       $device_config_file = get_rancid_filename($device['hostname']);

       // Print the config tab if we have a device config
-      if ($device_config_file)
+      if ($device_config_file || $oxidized_hostpath)
       {
         $navbar['options']['showconfig'] = array('text' => 'Config', 'icon' => $config['icon']['config']);
       }
--- observium-trunk/html/pages/device/showconfig.inc.php	2019-02-15 14:43:01.948652515 +1000
+++ root/html/pages/device/showconfig.inc.php	2019-03-13 22:34:08.510242794 +1000
@@ -80,6 +80,70 @@
     if ($rev['count']) { $rev['type'] = 'git'; }
   }

+  // Oxidized Integration
+  // NOTE: we check $rev['type'] to determine if a config has already been found via cvs or git, if so Oxidized will not be used for this device
+  if (!isset($rev['type'])
+    && isset($config['oxidized']['web']['enable'])
+    && $config['oxidized']['web']['enable']
+    && isset($config['oxidized']['web']['url'])
+    && $config['oxidized']['web']['url'] != '')
+  {
+    $rev_active_index = 0;
+
+    // do requests to Oxidized API
+    $context_data = array(
+      'method' => 'GET',
+      'header' =>
+        "Accept: application/json\r\n"
+    );
+
+    if (isset($config['oxidized']['web']['auth']['header'])
+        && isset($config['oxidized']['web']['auth']['value'])
+        && $config['oxidized']['web']['auth']['header'] != ''
+        && $config['oxidized']['web']['auth']['value'] != '')
+    {
+      $context_data['header'][] = $config['oxidized']['web']['auth']['header'] . ": " . $config['oxidized']['web']['auth']['value'] . "\r\n";
+    }
+
+    $url = $config['oxidized']['web']['url'] . '/node/show/' . $device['hostname'] . '?format=json';
+    $result = json_decode(get_http_request($url, $context_data), TRUE);
+
+    if (isset($result['full_name']) && strpos($device['hostname'], '.') != false)
+    {
+      // try again with hostname minus domain if included
+      $url = $config['oxidized']['web']['url'] . '/node/show/' . strtok($device['hostname'], '.') . '?format=json';
+      $result = json_decode(get_http_request($url, $context_data), TRUE);
+    }
+
+    if (isset($result['full_name']))
+    {
+      $oxidized_hostpath = $result['full_name'];
+    }
+
+    $url = $config['oxidized']['web']['url'] . '/node/fetch/' . $oxidized_hostpath;
+    $device_config = get_http_request($url, $context_data);
+    if ( $device_config != '' )
+    {
+      $rev['count']++;
+      $rev['type'] = 'oxidized';
+
+      $url = $config['oxidized']['web']['url'] . '/node/version?node_full=' . $oxidized_hostpath . '&format=json';
+      $result = json_decode(get_http_request($url, $context_data), TRUE);
+
+      if ($result != null && $result != [])
+      {
+        $rev['count'] = sizeof($result);
+
+        foreach ($result as $ox_rev)
+        {
+          $rev['list'][] = array('rev' => $ox_rev['oid'], 'date' => $ox_rev['date']);
+        }
+      }
+
+      unset($result);
+    }
+  }
+
   $navbar['options']['latest']['url']   = generate_url(array('page'=>'device','device'=>$device['device_id'],'tab'=>'showconfig'));
   $navbar['options']['latest']['class'] = 'active';
   if ($rev['count'])
@@ -94,9 +158,12 @@
     {
       $rev_max = 32;
     }
+    $rev_counter = $rev['count'];
     foreach ($rev['list'] as $i => $entry)
     {
+      $rev_counter--;
       $rev_name = ($rev['type'] == 'svn' ? 'r'.$entry['rev'] : $entry['rev']);
+      $rev_name = ($rev['type'] == 'oxidized' ? 'r'.$rev_counter : $rev_name);
       if ($i > ($rev_max - 1))
       {
         break; // Show only last 10 revisions
@@ -154,21 +221,35 @@
         $cmd_cat   = $config['git'].' --git-dir='. $git_dir .' --work-tree='.$cmd_dir.' show '.$rev['curr'].':'.$git_file;
         $cmd_diff  = $config['git'].' --git-dir='. $git_dir .' --work-tree='.$cmd_dir.' diff '.$rev['prev'].' '.$rev['curr'].' '.$cmd_file;
         $prev_name = $rev['prev'];
+      case 'oxidized':
+        $prev_name = 'r' . ($rev_active_index != 0 ? ($rev['count'] - $rev_active_index - 2) : $rev['count'] - 2);
     }
-    $device_config = external_exec($cmd_cat);
-    if (!isset($rev['prev']))
+    switch ($rev['type'])
     {
-      $diff = '';
-      if (empty($device_config))
-      {
-        $device_config = '# Initial device added.';
-      }
-    } else {
-      $diff = external_exec($cmd_diff);
-      if (!$diff)
-      {
-        $diff = 'No Difference';
-      }
+      case 'svn':
+      case 'git':
+        $device_config = external_exec($cmd_cat);
+        if (!isset($rev['prev']))
+        {
+          $diff = '';
+          if (empty($device_config))
+          {
+            $device_config = '# Initial device added.';
+          }
+        } else {
+          $diff = external_exec($cmd_diff);
+          if (!$diff)
+          {
+            $diff = 'No Difference';
+          }
+        }
+        break;
+      case 'oxidized':
+        $ox_date = urlencode($rev['list'][$rev_active_index]['date']);
+        $url = $config['oxidized']['web']['url'] . '/node/version/diffs?node=' . $device['hostname'] . '&group=' . $device['type'] . '&oid=' . $rev['curr'] . '&date=' . $ox_date . '&num=1&oid2=' . $rev['prev'] . '&format=json';
+        $diff = json_decode(get_http_request($url, $context_data), TRUE);
+        unset($ox_date);
+        break;
     }
   } else {
     $fh = fopen($device_config_file, 'r') or die("Can't open file");
